guake/dialogs.py

```diff
@@ -143,14 +143,24 @@ def __init__(self, terminal, window):
         self.parent_window = window
 
     def run(self):
-        self.terminal.select_all()
-        self.terminal.copy_clipboard()
-        self.terminal.unselect_all()
-        clipboard = Gtk.Clipboard.get_default(self.parent_window.get_display())
-        selection = clipboard.wait_for_text()
-        if not selection:
-            return
-        selection = selection.rstrip()
+        # row from get_cursor_position takes the scrollback rows into account automatically
+        row = self.terminal.get_cursor_position()[1]
+        # Get the max col, not where the cursor is, to avoid losing text at the end
+        col = self.terminal.get_column_count()
+        content = self.terminal.get_text_range(0, 0, row, col)
+
+        if content:
```

Comment:
I add this  suggestion to fix this error: https://github.com/Guake/guake/pull/2218#issuecomment-2134558574
```suggestion
        if content and content[0]:
```

Reply:
Looks like this does fix it.